libraries/primitive_consul_maintenance.rb

```diff
@@ -23,15 +23,30 @@ def register(choregraphie)
 
       choregraphie.before do
         require 'diplomat'
-        Diplomat::Maintenance.enable(true, @options[:reason])
+        checks = Diplomat::Agent.checks()
+        maint_status = checks.dig('_node_maintenance', 'Status')
+        if maint_status == 'critical'
```

Comment:
the following piece of code:
```
checks = Diplomat::Agent.checks()
maint_status = checks.dig('_node_maintenance', 'Status')
if maint_status == 'critical'
maint_notes = checks.dig('_node_maintenance', 'Notes')
```
is duplicated.

I think you could create a small method `current_maintenance` that look like:
```
def current_maintenance
  checks = Diplomat::Agent.checks()
  maint_status = checks.dig('_node_maintenance', 'Status')
  maint_status == 'critical' && checks.dig('_node_maintenance', 'Notes')
end
```

it returns false if there is no maintenance and a string with the reason if any.

then you can write code such as:
```
if current_maintenance
  Chef::Log.warn "Consul maintenance was already enabled, reason: #{current_maintenance}."
else
  Diplomat::Maintenance.enable(true, @options[:reason])
end
```